contracts/test/unit/VoltV2.t.sol

```diff
@@ -0,0 +1,384 @@
+//SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity =0.8.13;
+
+import {SafeCast} from "@openzeppelin/contracts/utils/math/SafeCast.sol";
+import {DSTest} from "../unit/utils/DSTest.sol";
+import {Vm} from "../unit/utils/Vm.sol";
+import {VoltV2} from "../../volt/VoltV2.sol";
+import {Core} from "../../core/Core.sol";
+import {getCore, getAddresses, VoltTestAddresses} from "./utils/Fixtures.sol";
+import {ICore} from "../../core/ICore.sol";
+import {stdError} from "../unit/utils/StdLib.sol";
+
+contract UnitTestVoltV2 is DSTest {
+    using SafeCast for *;
+    VoltV2 private volt;
+    ICore private core;
+    VoltTestAddresses public addresses = getAddresses();
+
+    Vm private vm = Vm(HEVM_ADDRESS);
+
+    function setUp() public {
+        core = getCore();
+        volt = new VoltV2(address(core));
+    }
+
```

Comment:
assert any transfer or mint of amount greater than uint224 causes an overflow safecast error.